// @ariakit/core/cjs/__chunks/A7K3KZJL.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});







var _A4GDJCZBcjs = require('./A4GDJCZB.cjs');


var _RCYVACJOcjs = require('./RCYVACJO.cjs');



var _72I2GWXFcjs = require('./72I2GWXF.cjs');

// src/disclosure/disclosure-store.ts
function createDisclosureStore(props = {}) {
  const store = _A4GDJCZBcjs.mergeStore.call(void 0, 
    props.store,
    _A4GDJCZBcjs.omit.call(void 0, props.disclosure, ["contentElement", "disclosureElement"])
  );
  _A4GDJCZBcjs.throwOnConflictingProps.call(void 0, props, store);
  const syncState = store == null ? void 0 : store.getState();
  const open = _RCYVACJOcjs.defaultValue.call(void 0, 
    props.open,
    syncState == null ? void 0 : syncState.open,
    props.defaultOpen,
    false
  );
  const animated = _RCYVACJOcjs.defaultValue.call(void 0, props.animated, syncState == null ? void 0 : syncState.animated, false);
  const initialState = {
    open,
    animated,
    animating: !!animated && open,
    mounted: open,
    contentElement: _RCYVACJOcjs.defaultValue.call(void 0, syncState == null ? void 0 : syncState.contentElement, null),
    disclosureElement: _RCYVACJOcjs.defaultValue.call(void 0, syncState == null ? void 0 : syncState.disclosureElement, null)
  };
  const disclosure = _A4GDJCZBcjs.createStore.call(void 0, initialState, store);
  _A4GDJCZBcjs.setup.call(void 0, 
    disclosure,
    () => _A4GDJCZBcjs.sync.call(void 0, disclosure, ["animated", "animating"], (state) => {
      if (state.animated)
        return;
      disclosure.setState("animating", false);
    })
  );
  _A4GDJCZBcjs.setup.call(void 0, 
    disclosure,
    () => _A4GDJCZBcjs.subscribe.call(void 0, disclosure, ["open"], () => {
      if (!disclosure.getState().animated)
        return;
      disclosure.setState("animating", true);
    })
  );
  _A4GDJCZBcjs.setup.call(void 0, 
    disclosure,
    () => _A4GDJCZBcjs.sync.call(void 0, disclosure, ["open", "animating"], (state) => {
      disclosure.setState("mounted", state.open || state.animating);
    })
  );
  return _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {}, disclosure), {
    setOpen: (value) => disclosure.setState("open", value),
    show: () => disclosure.setState("open", true),
    hide: () => disclosure.setState("open", false),
    toggle: () => disclosure.setState("open", (open2) => !open2),
    stopAnimation: () => disclosure.setState("animating", false),
    setContentElement: (value) => disclosure.setState("contentElement", value),
    setDisclosureElement: (value) => disclosure.setState("disclosureElement", value)
  });
}



exports.createDisclosureStore = createDisclosureStore;
