// @ariakit/react-core/cjs/form/form.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _VLI66TFLcjs = require('../__chunks/VLI66TFL.cjs');
require('../__chunks/NHZBX2RS.cjs');




var _Z5PUIG6Acjs = require('../__chunks/Z5PUIG6A.cjs');







var _5WLTMVVQcjs = require('../__chunks/5WLTMVVQ.cjs');
require('../__chunks/7WUTS26Y.cjs');




var _72I2GWXFcjs = require('../__chunks/72I2GWXF.cjs');

// src/form/form.tsx
var _react = require('react');
var _dom = require('@ariakit/core/utils/dom');
var _misc = require('@ariakit/core/utils/misc');
var _jsxruntime = require('react/jsx-runtime');
function isField(element, items) {
  return items.some(
    (item) => item.type === "field" && item.element === element
  );
}
function getFirstInvalidField(items) {
  return items.find(
    (item) => {
      var _a;
      return item.type === "field" && ((_a = item.element) == null ? void 0 : _a.getAttribute("aria-invalid")) === "true";
    }
  );
}
var useForm = _Z5PUIG6Acjs.createHook.call(void 0, 
  (_a) => {
    var _b = _a, {
      store,
      validateOnChange = true,
      validateOnBlur = true,
      resetOnUnmount = false,
      resetOnSubmit = true,
      autoFocusOnSubmit = true
    } = _b, props = _72I2GWXFcjs.__objRest.call(void 0, _b, [
      "store",
      "validateOnChange",
      "validateOnBlur",
      "resetOnUnmount",
      "resetOnSubmit",
      "autoFocusOnSubmit"
    ]);
    const context = _VLI66TFLcjs.useFormContext.call(void 0, );
    store = store || context;
    _misc.invariant.call(void 0, 
      store,
      process.env.NODE_ENV !== "production" && "Form must receive a `store` prop or be wrapped in a FormProvider component."
    );
    const ref = _react.useRef.call(void 0, null);
    const values = store.useState("values");
    const submitSucceed = store.useState("submitSucceed");
    const submitFailed = store.useState("submitFailed");
    const items = store.useState("items");
    const defaultValues = _5WLTMVVQcjs.useInitialValue.call(void 0, values);
    _react.useEffect.call(void 0, 
      () => resetOnUnmount ? store == null ? void 0 : store.reset : void 0,
      [resetOnUnmount, store]
    );
    _5WLTMVVQcjs.useUpdateEffect.call(void 0, () => {
      if (!validateOnChange)
        return;
      if (values === defaultValues)
        return;
      store == null ? void 0 : store.validate();
    }, [validateOnChange, values, defaultValues, store]);
    _react.useEffect.call(void 0, () => {
      if (!resetOnSubmit)
        return;
      if (!submitSucceed)
        return;
      store == null ? void 0 : store.reset();
    }, [resetOnSubmit, submitSucceed, store]);
    const [shouldFocusOnSubmit, setShouldFocusOnSubmit] = _react.useState.call(void 0, false);
    _react.useEffect.call(void 0, () => {
      if (!shouldFocusOnSubmit)
        return;
      if (!submitFailed)
        return;
      const field = getFirstInvalidField(items);
      const element = field == null ? void 0 : field.element;
      if (!element)
        return;
      setShouldFocusOnSubmit(false);
      element.focus();
      if (_dom.isTextField.call(void 0, element)) {
        element.select();
      }
    }, [autoFocusOnSubmit, submitFailed, items]);
    const onSubmitProp = props.onSubmit;
    const onSubmit = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      onSubmitProp == null ? void 0 : onSubmitProp(event);
      if (event.defaultPrevented)
        return;
      event.preventDefault();
      store == null ? void 0 : store.submit();
      if (!autoFocusOnSubmit)
        return;
      setShouldFocusOnSubmit(true);
    });
    const onBlurProp = props.onBlur;
    const onBlur = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      onBlurProp == null ? void 0 : onBlurProp(event);
      if (event.defaultPrevented)
        return;
      if (!validateOnBlur)
        return;
      if (!store)
        return;
      if (!isField(event.target, store.getState().items))
        return;
      store.validate();
    });
    const onResetProp = props.onReset;
    const onReset = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      onResetProp == null ? void 0 : onResetProp(event);
      if (event.defaultPrevented)
        return;
      event.preventDefault();
      store == null ? void 0 : store.reset();
    });
    props = _5WLTMVVQcjs.useWrapElement.call(void 0, 
      props,
      (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _VLI66TFLcjs.FormScopedContextProvider, { value: store, children: element }),
      [store]
    );
    const tagName = _5WLTMVVQcjs.useTagName.call(void 0, ref, props.as || "form");
    props = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      role: tagName !== "form" ? "form" : void 0,
      noValidate: true
    }, props), {
      ref: _5WLTMVVQcjs.useMergeRefs.call(void 0, ref, props.ref),
      onSubmit,
      onBlur,
      onReset
    });
    return props;
  }
);
var Form = _Z5PUIG6Acjs.createComponent.call(void 0, (props) => {
  const htmlProps = useForm(props);
  return _Z5PUIG6Acjs.createElement.call(void 0, "form", htmlProps);
});
if (process.env.NODE_ENV !== "production") {
  Form.displayName = "Form";
}



exports.Form = Form; exports.useForm = useForm;
