// @ariakit/react-core/cjs/__chunks/2H4Y2NEY.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _NER6RNCQcjs = require('./NER6RNCQ.cjs');


var _FO6MB5HFcjs = require('./FO6MB5HF.cjs');


var _HWZH54NBcjs = require('./HWZH54NB.cjs');




var _Z5PUIG6Acjs = require('./Z5PUIG6A.cjs');


var _EVNF7KMTcjs = require('./EVNF7KMT.cjs');





var _5WLTMVVQcjs = require('./5WLTMVVQ.cjs');




var _72I2GWXFcjs = require('./72I2GWXF.cjs');

// src/checkbox/checkbox.tsx
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
function setMixed(element, mixed) {
  if (mixed) {
    element.indeterminate = true;
  } else if (element.indeterminate) {
    element.indeterminate = false;
  }
}
function isNativeCheckbox(tagName, type) {
  return tagName === "input" && (!type || type === "checkbox");
}
function getNonArrayValue(value) {
  if (Array.isArray(value)) {
    return value.toString();
  }
  return value;
}
var useCheckbox = _Z5PUIG6Acjs.createHook.call(void 0, 
  (_a) => {
    var _b = _a, {
      store,
      value: valueProp,
      checked: checkedProp,
      defaultChecked
    } = _b, props = _72I2GWXFcjs.__objRest.call(void 0, _b, [
      "store",
      "value",
      "checked",
      "defaultChecked"
    ]);
    const context = _NER6RNCQcjs.useCheckboxContext.call(void 0, );
    store = store || context;
    const [_checked, setChecked] = _react.useState.call(void 0, defaultChecked != null ? defaultChecked : false);
    const checked = _EVNF7KMTcjs.useStoreState.call(void 0, store, (state) => {
      if (checkedProp !== void 0)
        return checkedProp;
      if ((state == null ? void 0 : state.value) === void 0)
        return _checked;
      if (valueProp != null) {
        if (Array.isArray(state.value)) {
          const nonArrayValue = getNonArrayValue(valueProp);
          return state.value.includes(nonArrayValue);
        }
        return state.value === valueProp;
      }
      if (Array.isArray(state.value))
        return false;
      if (typeof state.value === "boolean")
        return state.value;
      return false;
    });
    const ref = _react.useRef.call(void 0, null);
    const tagName = _5WLTMVVQcjs.useTagName.call(void 0, ref, props.as || "input");
    const nativeCheckbox = isNativeCheckbox(tagName, props.type);
    const mixed = checked ? checked === "mixed" : void 0;
    const isChecked = checked === "mixed" ? false : checked;
    _react.useEffect.call(void 0, () => {
      const element = ref.current;
      if (!element)
        return;
      setMixed(element, mixed);
      element.checked = isChecked;
    }, [mixed, isChecked]);
    const onChangeProp = props.onChange;
    const onChange = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      if (props.disabled) {
        event.stopPropagation();
        event.preventDefault();
        return;
      }
      setMixed(event.currentTarget, mixed);
      if (!nativeCheckbox) {
        event.currentTarget.checked = !event.currentTarget.checked;
      }
      onChangeProp == null ? void 0 : onChangeProp(event);
      if (event.defaultPrevented)
        return;
      const elementChecked = event.currentTarget.checked;
      setChecked(elementChecked);
      store == null ? void 0 : store.setValue((prevValue) => {
        if (valueProp == null)
          return elementChecked;
        const nonArrayValue = getNonArrayValue(valueProp);
        if (!Array.isArray(prevValue)) {
          return prevValue === nonArrayValue ? false : nonArrayValue;
        }
        if (elementChecked)
          return [...prevValue, nonArrayValue];
        return prevValue.filter((v) => v !== nonArrayValue);
      });
    });
    const onClickProp = props.onClick;
    const onClick = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      onClickProp == null ? void 0 : onClickProp(event);
      if (event.defaultPrevented)
        return;
      if (nativeCheckbox)
        return;
      onChange(event);
    });
    props = _5WLTMVVQcjs.useWrapElement.call(void 0, 
      props,
      (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _FO6MB5HFcjs.CheckboxCheckedContext.Provider, { value: isChecked, children: element }),
      [isChecked]
    );
    props = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      role: !nativeCheckbox ? "checkbox" : void 0,
      type: nativeCheckbox ? "checkbox" : void 0,
      "aria-checked": checked
    }, props), {
      ref: _5WLTMVVQcjs.useMergeRefs.call(void 0, ref, props.ref),
      onChange,
      onClick
    });
    props = _HWZH54NBcjs.useCommand.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, { clickOnEnter: !nativeCheckbox }, props));
    return _72I2GWXFcjs.__spreadValues.call(void 0, {
      value: nativeCheckbox ? valueProp : void 0,
      checked: isChecked
    }, props);
  }
);
var Checkbox = _Z5PUIG6Acjs.createComponent.call(void 0, (props) => {
  const htmlProps = useCheckbox(props);
  return _Z5PUIG6Acjs.createElement.call(void 0, "input", htmlProps);
});
if (process.env.NODE_ENV !== "production") {
  Checkbox.displayName = "Checkbox";
}




exports.useCheckbox = useCheckbox; exports.Checkbox = Checkbox;
