// @ariakit/react-core/cjs/__chunks/6RYSMHQY.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _H7R5D2FRcjs = require('./H7R5D2FR.cjs');



var _Q6W7IWNJcjs = require('./Q6W7IWNJ.cjs');


var _Z5PUIG6Acjs = require('./Z5PUIG6A.cjs');

// src/menu/menu-context.ts
var _react = require('react');
var menu = _Z5PUIG6Acjs.createStoreContext.call(void 0, 
  [_H7R5D2FRcjs.CompositeContextProvider, _Q6W7IWNJcjs.HovercardContextProvider],
  [_H7R5D2FRcjs.CompositeScopedContextProvider, _Q6W7IWNJcjs.HovercardScopedContextProvider]
);
var useMenuContext = menu.useContext;
var useMenuScopedContext = menu.useScopedContext;
var useMenuProviderContext = menu.useProviderContext;
var MenuContextProvider = menu.ContextProvider;
var MenuScopedContextProvider = menu.ScopedContextProvider;
var menubar = _Z5PUIG6Acjs.createStoreContext.call(void 0, 
  [_H7R5D2FRcjs.CompositeContextProvider],
  [_H7R5D2FRcjs.CompositeScopedContextProvider]
);
var useMenuBarContext = menubar.useContext;
var useMenuBarScopedContext = menubar.useScopedContext;
var useMenuBarProviderContext = menubar.useProviderContext;
var MenuBarContextProvider = menubar.ContextProvider;
var MenuBarScopedContextProvider = menubar.ScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);













exports.useMenuContext = useMenuContext; exports.useMenuScopedContext = useMenuScopedContext; exports.useMenuProviderContext = useMenuProviderContext; exports.MenuContextProvider = MenuContextProvider; exports.MenuScopedContextProvider = MenuScopedContextProvider; exports.useMenuBarContext = useMenuBarContext; exports.useMenuBarScopedContext = useMenuBarScopedContext; exports.useMenuBarProviderContext = useMenuBarProviderContext; exports.MenuBarContextProvider = MenuBarContextProvider; exports.MenuBarScopedContextProvider = MenuBarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
