// @ariakit/react-core/cjs/__chunks/CKEXTQ2S.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _KSFAO77Ecjs = require('./KSFAO77E.cjs');




var _Z5PUIG6Acjs = require('./Z5PUIG6A.cjs');



var _72I2GWXFcjs = require('./72I2GWXF.cjs');

// src/focus-trap/focus-trap.ts
var useFocusTrap = _Z5PUIG6Acjs.createHook.call(void 0, (props) => {
  props = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
    "data-focus-trap": "",
    tabIndex: 0,
    "aria-hidden": true
  }, props), {
    style: _72I2GWXFcjs.__spreadValues.call(void 0, {
      // Prevents unintended scroll jumps.
      position: "fixed",
      top: 0,
      left: 0
    }, props.style)
  });
  props = _KSFAO77Ecjs.useVisuallyHidden.call(void 0, props);
  return props;
});
var FocusTrap = _Z5PUIG6Acjs.createComponent.call(void 0, (props) => {
  const htmlProps = useFocusTrap(props);
  return _Z5PUIG6Acjs.createElement.call(void 0, "span", htmlProps);
});
if (process.env.NODE_ENV !== "production") {
  FocusTrap.displayName = "FocusTrap";
}




exports.useFocusTrap = useFocusTrap; exports.FocusTrap = FocusTrap;
