// @ariakit/react-core/cjs/__chunks/H7R5D2FR.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _NHZBX2RScjs = require('./NHZBX2RS.cjs');


var _Z5PUIG6Acjs = require('./Z5PUIG6A.cjs');

// src/composite/composite-context.tsx
var _react = require('react');
var ctx = _Z5PUIG6Acjs.createStoreContext.call(void 0, 
  [_NHZBX2RScjs.CollectionContextProvider],
  [_NHZBX2RScjs.CollectionScopedContextProvider]
);
var useCompositeContext = ctx.useContext;
var useCompositeScopedContext = ctx.useScopedContext;
var useCompositeProviderContext = ctx.useProviderContext;
var CompositeContextProvider = ctx.ContextProvider;
var CompositeScopedContextProvider = ctx.ScopedContextProvider;
var CompositeItemContext = _react.createContext.call(void 0, 
  void 0
);
var CompositeRowContext = _react.createContext.call(void 0, 
  void 0
);









exports.useCompositeContext = useCompositeContext; exports.useCompositeScopedContext = useCompositeScopedContext; exports.useCompositeProviderContext = useCompositeProviderContext; exports.CompositeContextProvider = CompositeContextProvider; exports.CompositeScopedContextProvider = CompositeScopedContextProvider; exports.CompositeItemContext = CompositeItemContext; exports.CompositeRowContext = CompositeRowContext;
