// @ariakit/react-core/cjs/__chunks/I2EPCLS6.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _H7R5D2FRcjs = require('./H7R5D2FR.cjs');



var _AKOG5S2Tcjs = require('./AKOG5S2T.cjs');


var _Z5PUIG6Acjs = require('./Z5PUIG6A.cjs');

// src/select/select-context.tsx
var _react = require('react');
var ctx = _Z5PUIG6Acjs.createStoreContext.call(void 0, 
  [_AKOG5S2Tcjs.PopoverContextProvider, _H7R5D2FRcjs.CompositeContextProvider],
  [_AKOG5S2Tcjs.PopoverScopedContextProvider, _H7R5D2FRcjs.CompositeScopedContextProvider]
);
var useSelectContext = ctx.useContext;
var useSelectScopedContext = ctx.useScopedContext;
var useSelectProviderContext = ctx.useProviderContext;
var SelectContextProvider = ctx.ContextProvider;
var SelectScopedContextProvider = ctx.ScopedContextProvider;
var SelectItemCheckedContext = _react.createContext.call(void 0, false);








exports.useSelectContext = useSelectContext; exports.useSelectScopedContext = useSelectScopedContext; exports.useSelectProviderContext = useSelectProviderContext; exports.SelectContextProvider = SelectContextProvider; exports.SelectScopedContextProvider = SelectScopedContextProvider; exports.SelectItemCheckedContext = SelectItemCheckedContext;
