// @ariakit/react-core/cjs/__chunks/L2VQ4YLN.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _W6ZRSMO5cjs = require('./W6ZRSMO5.cjs');


var _A2NMYLCGcjs = require('./A2NMYLCG.cjs');




var _Z5PUIG6Acjs = require('./Z5PUIG6A.cjs');


var _EVNF7KMTcjs = require('./EVNF7KMT.cjs');





var _5WLTMVVQcjs = require('./5WLTMVVQ.cjs');




var _72I2GWXFcjs = require('./72I2GWXF.cjs');

// src/disclosure/disclosure-content.tsx
var _react = require('react');
var _misc = require('@ariakit/core/utils/misc');
var _jsxruntime = require('react/jsx-runtime');
function afterTimeout(timeoutMs, cb) {
  const timeoutId = setTimeout(cb, timeoutMs);
  return () => clearTimeout(timeoutId);
}
function afterPaint(cb) {
  let raf = requestAnimationFrame(() => {
    raf = requestAnimationFrame(cb);
  });
  return () => cancelAnimationFrame(raf);
}
function parseCSSTime(...times) {
  return times.join(", ").split(", ").reduce((longestTime, currentTimeString) => {
    const currentTime = parseFloat(currentTimeString || "0s") * 1e3;
    if (currentTime > longestTime)
      return currentTime;
    return longestTime;
  }, 0);
}
function isHidden(mounted, hidden, alwaysVisible) {
  return !alwaysVisible && hidden !== false && (!mounted || !!hidden);
}
var useDisclosureContent = _Z5PUIG6Acjs.createHook.call(void 0, 
  (_a) => {
    var _b = _a, { store, alwaysVisible } = _b, props = _72I2GWXFcjs.__objRest.call(void 0, _b, ["store", "alwaysVisible"]);
    const context = _A2NMYLCGcjs.useDisclosureProviderContext.call(void 0, );
    store = store || context;
    _misc.invariant.call(void 0, 
      store,
      process.env.NODE_ENV !== "production" && "DisclosureContent must receive a `store` prop or be wrapped in a DisclosureProvider component."
    );
    const id = _5WLTMVVQcjs.useId.call(void 0, props.id);
    const [transition, setTransition] = _react.useState.call(void 0, null);
    const open = store.useState("open");
    const mounted = store.useState("mounted");
    const animated = store.useState("animated");
    const contentElement = store.useState("contentElement");
    _5WLTMVVQcjs.useSafeLayoutEffect.call(void 0, () => {
      if (!animated)
        return;
      if (!(contentElement == null ? void 0 : contentElement.isConnected)) {
        setTransition(null);
        return;
      }
      return afterPaint(() => {
        setTransition(open ? "enter" : "leave");
      });
    }, [animated, contentElement, open]);
    _5WLTMVVQcjs.useSafeLayoutEffect.call(void 0, () => {
      if (!store)
        return;
      if (!animated)
        return;
      if (!contentElement)
        return;
      if (!transition)
        return;
      if (transition === "enter" && !open)
        return;
      if (transition === "leave" && open)
        return;
      if (typeof animated === "number") {
        const timeoutMs2 = animated;
        return afterTimeout(timeoutMs2, store.stopAnimation);
      }
      const {
        transitionDuration,
        animationDuration,
        transitionDelay,
        animationDelay
      } = getComputedStyle(contentElement);
      const delay = parseCSSTime(transitionDelay, animationDelay);
      const duration = parseCSSTime(transitionDuration, animationDuration);
      const timeoutMs = delay + duration;
      if (!timeoutMs)
        return;
      return afterTimeout(timeoutMs, store.stopAnimation);
    }, [store, animated, contentElement, open, transition]);
    props = _5WLTMVVQcjs.useWrapElement.call(void 0, 
      props,
      (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _W6ZRSMO5cjs.DialogScopedContextProvider, { value: store, children: element }),
      [store]
    );
    const hidden = isHidden(mounted, props.hidden, alwaysVisible);
    const style = hidden ? _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {}, props.style), { display: "none" }) : props.style;
    props = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      id,
      "data-enter": transition === "enter" ? "" : void 0,
      "data-leave": transition === "leave" ? "" : void 0,
      hidden
    }, props), {
      ref: _5WLTMVVQcjs.useMergeRefs.call(void 0, id ? store.setContentElement : null, props.ref),
      style
    });
    return props;
  }
);
var DisclosureContentImpl = _Z5PUIG6Acjs.createComponent.call(void 0, 
  (props) => {
    const htmlProps = useDisclosureContent(props);
    return _Z5PUIG6Acjs.createElement.call(void 0, "div", htmlProps);
  }
);
var DisclosureContent = _Z5PUIG6Acjs.createComponent.call(void 0, 
  (_a) => {
    var _b = _a, { unmountOnHide } = _b, props = _72I2GWXFcjs.__objRest.call(void 0, _b, ["unmountOnHide"]);
    const context = _A2NMYLCGcjs.useDisclosureProviderContext.call(void 0, );
    const store = props.store || context;
    const mounted = _EVNF7KMTcjs.useStoreState.call(void 0, 
      store,
      (state) => !unmountOnHide || (state == null ? void 0 : state.mounted)
    );
    if (mounted === false)
      return null;
    return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, DisclosureContentImpl, _72I2GWXFcjs.__spreadValues.call(void 0, {}, props));
  }
);
if (process.env.NODE_ENV !== "production") {
  DisclosureContent.displayName = "DisclosureContent";
}





exports.isHidden = isHidden; exports.useDisclosureContent = useDisclosureContent; exports.DisclosureContent = DisclosureContent;
