// @ariakit/react-core/cjs/__chunks/ZBG4TGVL.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _H7R5D2FRcjs = require('./H7R5D2FR.cjs');



var _AKOG5S2Tcjs = require('./AKOG5S2T.cjs');


var _Z5PUIG6Acjs = require('./Z5PUIG6A.cjs');

// src/combobox/combobox-context.tsx
var _react = require('react');
var ctx = _Z5PUIG6Acjs.createStoreContext.call(void 0, 
  [_AKOG5S2Tcjs.PopoverContextProvider, _H7R5D2FRcjs.CompositeContextProvider],
  [_AKOG5S2Tcjs.PopoverScopedContextProvider, _H7R5D2FRcjs.CompositeScopedContextProvider]
);
var useComboboxContext = ctx.useContext;
var useComboboxScopedContext = ctx.useScopedContext;
var useComboboxProviderContext = ctx.useProviderContext;
var ComboboxContextProvider = ctx.ContextProvider;
var ComboboxScopedContextProvider = ctx.ScopedContextProvider;
var ComboboxItemValueContext = _react.createContext.call(void 0, 
  void 0
);








exports.useComboboxContext = useComboboxContext; exports.useComboboxScopedContext = useComboboxScopedContext; exports.useComboboxProviderContext = useComboboxProviderContext; exports.ComboboxContextProvider = ComboboxContextProvider; exports.ComboboxScopedContextProvider = ComboboxScopedContextProvider; exports.ComboboxItemValueContext = ComboboxItemValueContext;
