// @ariakit/react-core/cjs/select/select-item.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _LPFT4ZFHcjs = require('../__chunks/LPFT4ZFH.cjs');



var _I2EPCLS6cjs = require('../__chunks/I2EPCLS6.cjs');


var _GUWJ7VTOcjs = require('../__chunks/GUWJ7VTO.cjs');
require('../__chunks/HWZH54NB.cjs');
require('../__chunks/FMI5QF7N.cjs');
require('../__chunks/KLZID3QO.cjs');
require('../__chunks/H7R5D2FR.cjs');
require('../__chunks/NHZBX2RS.cjs');
require('../__chunks/AKOG5S2T.cjs');
require('../__chunks/W6ZRSMO5.cjs');
require('../__chunks/A2NMYLCG.cjs');
require('../__chunks/GQAB4M3B.cjs');
require('../__chunks/LHHW5ZQP.cjs');




var _Z5PUIG6Acjs = require('../__chunks/Z5PUIG6A.cjs');
require('../__chunks/EVNF7KMT.cjs');





var _5WLTMVVQcjs = require('../__chunks/5WLTMVVQ.cjs');
require('../__chunks/7WUTS26Y.cjs');




var _72I2GWXFcjs = require('../__chunks/72I2GWXF.cjs');

// src/select/select-item.tsx
var _react = require('react');
var _dom = require('@ariakit/core/utils/dom');
var _events = require('@ariakit/core/utils/events');
var _misc = require('@ariakit/core/utils/misc');
var _jsxruntime = require('react/jsx-runtime');
function isSelected(storeValue, itemValue) {
  if (storeValue == null)
    return false;
  if (itemValue == null)
    return false;
  if (Array.isArray(storeValue)) {
    return storeValue.includes(itemValue);
  }
  return storeValue === itemValue;
}
var useSelectItem = _Z5PUIG6Acjs.createHook.call(void 0, 
  (_a) => {
    var _b = _a, {
      store,
      value,
      getItem: getItemProp,
      hideOnClick,
      setValueOnClick = value != null,
      preventScrollOnKeyDown = true,
      focusOnHover = true
    } = _b, props = _72I2GWXFcjs.__objRest.call(void 0, _b, [
      "store",
      "value",
      "getItem",
      "hideOnClick",
      "setValueOnClick",
      "preventScrollOnKeyDown",
      "focusOnHover"
    ]);
    var _a2;
    const context = _I2EPCLS6cjs.useSelectScopedContext.call(void 0, );
    store = store || context;
    _misc.invariant.call(void 0, 
      store,
      process.env.NODE_ENV !== "production" && "SelectItem must be wrapped in a SelectList or SelectPopover component."
    );
    const id = _5WLTMVVQcjs.useId.call(void 0, props.id);
    const disabled = props.disabled;
    const getItem = _react.useCallback.call(void 0, 
      (item) => {
        const nextItem = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {}, item), {
          value: disabled ? void 0 : value,
          children: value
        });
        if (getItemProp) {
          return getItemProp(nextItem);
        }
        return nextItem;
      },
      [disabled, value, getItemProp]
    );
    const multiSelectable = store.useState(
      (state) => Array.isArray(state.value)
    );
    hideOnClick = hideOnClick != null ? hideOnClick : value != null && !multiSelectable;
    const onClickProp = props.onClick;
    const setValueOnClickProp = _5WLTMVVQcjs.useBooleanEvent.call(void 0, setValueOnClick);
    const hideOnClickProp = _5WLTMVVQcjs.useBooleanEvent.call(void 0, hideOnClick);
    const onClick = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      onClickProp == null ? void 0 : onClickProp(event);
      if (event.defaultPrevented)
        return;
      if (_events.isDownloading.call(void 0, event))
        return;
      if (_events.isOpeningInNewTab.call(void 0, event))
        return;
      if (setValueOnClickProp(event) && value != null) {
        store == null ? void 0 : store.setValue((prevValue) => {
          if (!Array.isArray(prevValue))
            return value;
          if (prevValue.includes(value)) {
            return prevValue.filter((v) => v !== value);
          }
          return [...prevValue, value];
        });
      }
      if (hideOnClickProp(event)) {
        store == null ? void 0 : store.hide();
      }
    });
    const selected = store.useState((state) => isSelected(state.value, value));
    props = _5WLTMVVQcjs.useWrapElement.call(void 0, 
      props,
      (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _I2EPCLS6cjs.SelectItemCheckedContext.Provider, { value: selected, children: element }),
      [selected]
    );
    const contentElement = store.useState("contentElement");
    const autoFocus = store.useState((state) => {
      if (state.activeId !== id && (store == null ? void 0 : store.item(state.activeId)))
        return false;
      if (state.value == null)
        return false;
      if (value == null)
        return false;
      if (Array.isArray(state.value)) {
        return state.value[state.value.length - 1] === value;
      }
      return state.value === value;
    });
    props = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      id,
      role: _dom.getPopupItemRole.call(void 0, contentElement),
      "aria-selected": selected,
      children: value
    }, props), {
      autoFocus: (_a2 = props.autoFocus) != null ? _a2 : autoFocus,
      onClick
    });
    props = _GUWJ7VTOcjs.useCompositeItem.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      store,
      getItem,
      preventScrollOnKeyDown
    }, props));
    const focusOnHoverProp = _5WLTMVVQcjs.useBooleanEvent.call(void 0, focusOnHover);
    props = _LPFT4ZFHcjs.useCompositeHover.call(void 0, _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      store
    }, props), {
      // We have to disable focusOnHover when the popup is closed, otherwise
      // the active item will change to null (the container) when the popup is
      // closed by clicking on an item.
      focusOnHover: (event) => {
        if (!focusOnHoverProp(event))
          return false;
        const state = store == null ? void 0 : store.getState();
        return !!(state == null ? void 0 : state.open);
      }
    }));
    return props;
  }
);
var SelectItem = _Z5PUIG6Acjs.createMemoComponent.call(void 0, (props) => {
  const htmlProps = useSelectItem(props);
  return _Z5PUIG6Acjs.createElement.call(void 0, "div", htmlProps);
});
if (process.env.NODE_ENV !== "production") {
  SelectItem.displayName = "SelectItem";
}



exports.SelectItem = SelectItem; exports.useSelectItem = useSelectItem;
