// @ariakit/react-core/cjs/combobox/combobox-item.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _LPFT4ZFHcjs = require('../__chunks/LPFT4ZFH.cjs');



var _ZBG4TGVLcjs = require('../__chunks/ZBG4TGVL.cjs');


var _GUWJ7VTOcjs = require('../__chunks/GUWJ7VTO.cjs');
require('../__chunks/HWZH54NB.cjs');
require('../__chunks/FMI5QF7N.cjs');
require('../__chunks/KLZID3QO.cjs');
require('../__chunks/H7R5D2FR.cjs');
require('../__chunks/NHZBX2RS.cjs');
require('../__chunks/AKOG5S2T.cjs');
require('../__chunks/W6ZRSMO5.cjs');
require('../__chunks/A2NMYLCG.cjs');
require('../__chunks/GQAB4M3B.cjs');
require('../__chunks/LHHW5ZQP.cjs');




var _Z5PUIG6Acjs = require('../__chunks/Z5PUIG6A.cjs');
require('../__chunks/EVNF7KMT.cjs');




var _5WLTMVVQcjs = require('../__chunks/5WLTMVVQ.cjs');
require('../__chunks/7WUTS26Y.cjs');




var _72I2GWXFcjs = require('../__chunks/72I2GWXF.cjs');

// src/combobox/combobox-item.tsx
var _react = require('react');
var _dom = require('@ariakit/core/utils/dom');
var _events = require('@ariakit/core/utils/events');
var _focus = require('@ariakit/core/utils/focus');
var _misc = require('@ariakit/core/utils/misc');
var _jsxruntime = require('react/jsx-runtime');
var useComboboxItem = _Z5PUIG6Acjs.createHook.call(void 0, 
  (_a) => {
    var _b = _a, {
      store,
      value,
      hideOnClick = value != null,
      setValueOnClick = true,
      focusOnHover = false,
      moveOnKeyPress = true,
      getItem: getItemProp
    } = _b, props = _72I2GWXFcjs.__objRest.call(void 0, _b, [
      "store",
      "value",
      "hideOnClick",
      "setValueOnClick",
      "focusOnHover",
      "moveOnKeyPress",
      "getItem"
    ]);
    const context = _ZBG4TGVLcjs.useComboboxScopedContext.call(void 0, );
    store = store || context;
    _misc.invariant.call(void 0, 
      store,
      process.env.NODE_ENV !== "production" && "ComboboxItem must be wrapped in a ComboboxList or ComboboxPopover component."
    );
    const getItem = _react.useCallback.call(void 0, 
      (item) => {
        const nextItem = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {}, item), { value });
        if (getItemProp) {
          return getItemProp(nextItem);
        }
        return nextItem;
      },
      [value, getItemProp]
    );
    const onClickProp = props.onClick;
    const setValueOnClickProp = _5WLTMVVQcjs.useBooleanEvent.call(void 0, setValueOnClick);
    const hideOnClickProp = _5WLTMVVQcjs.useBooleanEvent.call(void 0, hideOnClick);
    const onClick = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      onClickProp == null ? void 0 : onClickProp(event);
      if (event.defaultPrevented)
        return;
      if (_events.isDownloading.call(void 0, event))
        return;
      if (_events.isOpeningInNewTab.call(void 0, event))
        return;
      if (value != null && setValueOnClickProp(event)) {
        store == null ? void 0 : store.setValue(value);
      }
      if (hideOnClickProp(event)) {
        store == null ? void 0 : store.move(null);
        store == null ? void 0 : store.hide();
      }
    });
    const onKeyDownProp = props.onKeyDown;
    const onKeyDown = _5WLTMVVQcjs.useEvent.call(void 0, (event) => {
      onKeyDownProp == null ? void 0 : onKeyDownProp(event);
      if (event.defaultPrevented)
        return;
      const baseElement = store == null ? void 0 : store.getState().baseElement;
      if (!baseElement)
        return;
      if (_focus.hasFocus.call(void 0, baseElement))
        return;
      const printable = event.key.length === 1;
      if (printable || event.key === "Backspace" || event.key === "Delete") {
        queueMicrotask(() => baseElement.focus());
        if (_dom.isTextField.call(void 0, baseElement)) {
          store == null ? void 0 : store.setValue(baseElement.value);
        }
      }
    });
    props = _5WLTMVVQcjs.useWrapElement.call(void 0, 
      props,
      (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _ZBG4TGVLcjs.ComboboxItemValueContext.Provider, { value, children: element }),
      [value]
    );
    const contentElement = store.useState("contentElement");
    props = _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      role: _dom.getPopupItemRole.call(void 0, contentElement),
      children: value
    }, props), {
      onClick,
      onKeyDown
    });
    const moveOnKeyPressProp = _5WLTMVVQcjs.useBooleanEvent.call(void 0, moveOnKeyPress);
    props = _GUWJ7VTOcjs.useCompositeItem.call(void 0, _72I2GWXFcjs.__spreadProps.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, {
      store
    }, props), {
      getItem,
      // Dispatch a custom event on the combobox input when moving to an item
      // with the keyboard so the Combobox component can enable inline
      // autocompletion.
      moveOnKeyPress: (event) => {
        if (!moveOnKeyPressProp(event))
          return false;
        const moveEvent = new Event("combobox-item-move");
        const baseElement = store == null ? void 0 : store.getState().baseElement;
        baseElement == null ? void 0 : baseElement.dispatchEvent(moveEvent);
        return true;
      }
    }));
    props = _LPFT4ZFHcjs.useCompositeHover.call(void 0, _72I2GWXFcjs.__spreadValues.call(void 0, { store, focusOnHover }, props));
    return props;
  }
);
var ComboboxItem = _Z5PUIG6Acjs.createMemoComponent.call(void 0, 
  (props) => {
    const htmlProps = useComboboxItem(props);
    return _Z5PUIG6Acjs.createElement.call(void 0, "div", htmlProps);
  }
);
if (process.env.NODE_ENV !== "production") {
  ComboboxItem.displayName = "ComboboxItem";
}



exports.ComboboxItem = ComboboxItem; exports.useComboboxItem = useComboboxItem;
